### PR TITLE
Parse private decl

### LINKED ISSUE #679: Parser error?
```
org.aya.generic.util.InternalException: DECL: private def diag (a : Nat) : Type
| 0 => Nat
| suc a => False
	at org.aya.cli.parse.AyaGKProducer.unreachable(AyaGKProducer.java:870)
	at org.aya.cli.parse.AyaGKProducer.stmt(AyaGKProducer.java:105)
	at kala.collection.internal.view.SeqViews$FlatMapped.lambda$iterator$0(SeqViews.java:1142)
	at kala.collection.base.Iterators$3.next(Iterators.java:777)
	at kala.collection.base.Iterators$ConcatAll.hasNext(Iterators.java:2116)
	at kala.collection.immutable.ImmutableVector.from(ImmutableVector.java:159)
	at kala.collection.immutable.ImmutableVector.from(ImmutableVector.java:140)
	at kala.collection.CollectionLike.toImmutableVector(CollectionLike.java:107)
	at kala.collection.CollectionLike.toImmutableSeq(CollectionLike.java:95)
	at org.aya.cli.parse.AyaGKProducer.program(AyaGKProducer.java:86)
	at org.aya.cli.parse.AyaParserImpl.parse(AyaParserImpl.java:58)
	at org.aya.cli.parse.AyaParserImpl.program(AyaParserImpl.java:48)
	at org.aya.concrete.GenericAyaParser.program(GenericAyaParser.java:21)
	at org.aya.cli.library.LibraryCompiler.parse(LibraryCompiler.java:95)
	at org.aya.cli.library.LibraryCompiler.parseIfNeeded(LibraryCompiler.java:102)
	at org.aya.cli.library.LibraryCompiler.resolveImportsIfNeeded(LibraryCompiler.java:112)
	at org.aya.cli.library.LibraryCompiler.lambda$resolveImports$3(LibraryCompiler.java:129)
	at kala.function.CheckedConsumer.accept(CheckedConsumer.java:46)
	at kala.collection.IndexedSeqLike.forEach(IndexedSeqLike.java:815)
	at kala.collection.internal.view.CollectionViews$Of.forEach(CollectionViews.java:446)
	at kala.collection.base.Traversable.forEachChecked(Traversable.java:706)
	at org.aya.cli.library.LibraryCompiler.resolveImports(LibraryCompiler.java:128)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:194)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:182)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:161)
	at org.aya.cli.utils.AyaCompiler.catching(AyaCompiler.java:30)
	at org.aya.cli.library.LibraryCompiler.start(LibraryCompiler.java:147)
	at org.aya.lsp.server.AyaLanguageServer.loadLibrary(AyaLanguageServer.java:201)
	at kala.collection.internal.view.SeqViews$FlatMapped.lambda$iterator$0(SeqViews.java:1142)
	at kala.collection.base.Iterators$3.next(Iterators.java:777)
	at kala.collection.base.Iterators$ConcatAll.hasNext(Iterators.java:2116)
	at kala.collection.immutable.ImmutableVector.from(ImmutableVector.java:155)
	at kala.collection.immutable.ImmutableVector.from(ImmutableVector.java:140)
	at kala.collection.CollectionLike.toImmutableVector(CollectionLike.java:107)
	at kala.collection.CollectionLike.toImmutableSeq(CollectionLike.java:95)
	at org.aya.lsp.server.AyaLanguageServer.reload(AyaLanguageServer.java:192)
	at org.aya.lsp.server.AyaLanguageServer.load(AyaLanguageServer.java:373)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.javacs.lsp.LSP.connect(LSP.java:244)
	at org.aya.lsp.LspMain.call(LspMain.java:35)
	at org.aya.lsp.LspMain.call(LspMain.java:22)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.aya.lsp.LspMain.main(LspMain.java:24)
```

### PR DESCRIPTION
**Pull request by imkiva**
fix #679

### CODE REVIEW

File: cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java [hunk 1]
```diff
@@ -90,8 +90,11 @@ public record AyaGKProducer(
     if (node.is(IMPORT_CMD)) return ImmutableSeq.of(importCmd(node));
     if (node.is(MODULE)) return ImmutableSeq.of(module(node));
     if (node.is(OPEN_CMD)) return openCmd(node);
-    if (node.is(DECL)) {
-      var result = decl(node);
+    if (node.is(DECL) || node.peekChild(KW_PRIVATE) != null) {
```
Comment (HoshinoTented) [inline cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:93]: When do we get a `private something` but not a `private decl` ?
Are there any problems if we use `node.is(DECL)` ?

Comment (imkiva) [reply cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:93]: `node.is(DECL)` is like `node instanceOf Decl` where `Decl` has 4 subclasses `Fn`, `Struct`, `Data` and `Prim`.
Similarly, `DECL` is a TokenSet of `FN`, `STRUCT`, `DATA` and `PRIM`

But GK doesn't parse `KW_PRIVATE? decl` to any node of type `FN`, `STRUCT`, `DATA` or `PRIM`, whenever the `KW_PRIVATE` is present and matched, instead, it produces a `DECL`-node.

We of course could add `AyaPsiElementTypes.DECL` to the `DECL` TokenSet, but we will need an extra `node = isPrivate ? node.child(DECL) : node` in either way.

Comment (imkiva) [reply cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:93]: See the new proposal #690 which may address this issue in a more mature way.

Comment (HoshinoTented) [reply cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:93]: I see, thanks for your explanation.

Comment (imkiva) [reply cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:93]: You're right! I am moving the private check to `decl`'s producer.

File: cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public record AyaGKProducer(
   public static final @NotNull TokenSet ARGUMENT = AyaPsiParser.EXTENDS_SETS_[2];
   public static final @NotNull TokenSet STMT = AyaPsiParser.EXTENDS_SETS_[3];
   public static final @NotNull TokenSet EXPR = AyaPsiParser.EXTENDS_SETS_[4];
-  public static final @NotNull TokenSet DECL = TokenSet.create(DATA_DECL, FN_DECL, PRIM_DECL, STRUCT_DECL);
+  public static final @NotNull TokenSet DECL = TokenSet.create(AyaPsiElementTypes.DECL, DATA_DECL, FN_DECL, PRIM_DECL, STRUCT_DECL);
```
Comment (ice1000) [inline cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:81]: Why is this?

**Review by ice1000 — COMMENTED** [review]
🤔

File: cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java (hunk 2 above)
Comment (ice1000) [reply cli/src/main/java/org/aya/cli/parse/AyaGKProducer.java:81]: Ok I see why now

**Pull request merged**